kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/informers/InformerExceptionHandler.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.informers;
+
+import io.fabric8.kubernetes.client.WatcherException;
+
+public interface InformerExceptionHandler {
+
+  /**
+   * Provides a callback when the informer could terminated with a non-recoverable exception.
+   *
+   * @param t the {@link Throwable}, which may occur as either the cause of a non-http gone {@link WatcherException} or an
+   *        exception from calling list or watch
+   * @return true if the informer should retry, false if it should stop
+   */
+  boolean retry(Throwable t);
```

Comment:
nit: `retry` sounds imperative and might confuse users, what do you think of renaming to `shouldRetry`?

Also, it's unclear what the informer is retrying, maybe `shouldReconnect` would be more accurate.
`returns true if the Informer should attempt to reconnect after receiving the provided Throwable`

Reply:
> nit: retry sounds imperative and might confuse users, what do you think of renaming to shouldRetry?

Sure, I'll use shouldRetry instead.

> Also, it's unclear what the informer is retrying

That should be clear from the main description and the parameter description.

> maybe shouldReconnect would be more accurate.

I'm not sure about that.  We were using the term reconnect internally, but that's short-hand for listSyncAndWatch as it's not just the watch that needs re-established.